Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/UIMaterialInstantiator.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    public class UIMaterialInstiator
+    {
+        // this set ensures that we do not end up creating multiple copies of materials for every theme targetting the same instance
+        private static HashSet<int> targetInstances = new HashSet<int>();
+
+        public static void TryCreateMaterialCopy(ref Graphic targetGraphic)
```

Comment:
>TryCreateMaterialCopy [](start = 27, length = 21)

Instead of a void function, you could have this return the material and then just make it an extension method of Graphic.

public static Material GetNonSharedMaterial(this Graphic @this) {}
   or GetUniqueMaterial() ?  I'm up for better name suggestions.

Then usage would be like this:

Graphic myGraphic = whatever();
myGraphic.GetNonSharedMaterial().SetColor(propId, newColor);

Reply:
i had considered that.. but that would also mean that all "graphic.material" calls be replaced by  "graphic.getuniquematerial()/" and  people could forget to so that.